interpreter/interpreter.go

```diff
@@ -151,6 +156,230 @@ func CompileRegexConstants(regexOptimizations ...*RegexOptimization) Interpretab
 	return decRegexOptimizer(regexOptimizations...)
 }
 
+type typeVerifier func(int64, ...*types.TypeValue) (bool, error)
```

Comment:
Is there any reason we couldn't move this up into the `ProgramOptions()` for the `ext.Strings()` library?

Reply:
I believe the function `InterpolateFormattedString` doesn't have any internal deps on unexported interpreter functions and types now, so it should be possible to move it into `ext/strings`

Reply:
Ah, I see what you're getting at. You need the AST reference, or some way to resolve type information given an id ... let me think about it. It's very close to being something that can be self contained. Ideally, the verification would happen right after type-check as a new `EnvOption`

Reply:
(marking as resolved due to offline discussion to leave the code organized as-is for now, but definitely feel free to comment/unresolve)